packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -142,7 +146,7 @@ export const useGridSelection = (apiRef: GridApiRef, props: GridComponentProps):
 
   const setSelectionModel = React.useCallback(
     (model: GridSelectionModel) => {
-      apiRef.current.selectRows(model, true, true);
+      apiRef.current.selectRows(getArraySelectionModel(model), true, true);
```

Comment:
Why do we need this change here?

Reply:
I can edit `selectRows` to handle `GridRowId | GridRowId[]` otherwise

Reply:
https://github.com/mui-org/material-ui-x/pull/2176/files#diff-36939e58755c91e3693a4f18aa7796b137ba1e32f65357a940fb358286048088L61

On `master` we are doing this.
Which is wrong according to the typing because `selectionModel` is always an array in the interface.

https://github.com/mui-org/material-ui-x/blob/fa346f0fbe3d9b9eea9bb403fe4675f544d6abf9/packages/grid/_modules_/grid/models/gridSelectionModel.ts

I'm fixing this by supporting the non-array format.

Then I can either preprocess the prop like it was done in the DataGrid but in both components.
Or handle the non-array version across the codebase.
Which has the advantage of sticking to what the user is giving me. But that's debatable.

Reply:
The test are not different
I just added one for the array version in DataGrid.

Reply:
I cleaned it to only convert the non-array to array once from the props and once from the imperative update.

Reply:
@oliviertassinari  I just saw that the `setSelectionModel` does not handle the non-array version
Do we want to add it for consistency with the props or do we want to only handle the array version here ?

If so, I can remove the line you pointed. 

The issue came from the fact that the hook method it not typed by importing the public API
So we have

```ts
  const setSelectionModel = React.useCallback(
    (model: GridSelectionModel) => {
```

But

```ts
  /**
   * Updates the selected rows to be those passed to the `rowIds` argument.
   * Any row already selected will be unselected.
   * @param {GridRowId[]} rowIds The row ids to select.
   */
  setSelectionModel: (rowIds: GridRowId[]) => void;
```

Which made me make that mistake.

____

I removed the non-array version from `setSelectionModel` since it was not documented and it's not the subject of the PR to improve it :+1: